### DISCUSSION TITLE
ll_to_earth errors with whole number lat and long in raw query

### DISCUSSION THREAD

**Discussion by aeddie-zapidhire** [prisma/prisma#10679] [Q&A] [answered]
Have a strange problem with parameter replacement. Consider a dataset that include a `latitude` and `longitude` column. I can query this directly in DBeaver like this:

```sql
select
  earth_distance(ll_to_earth(1, 1), ll_to_earth(latitude, longitude)) AS distance
  , latitude 
  , longitude 
  , id
from thing
```

Runs fine. Note whole number `1`'s are passed to `ll_to_earth`.

Now, crafting a parameterised query in a Node.js app like this:

```js
const lat = 1
const long = 1
const results = await this.db.$queryRaw<any[]>`
select
  earth_distance(ll_to_earth(${+lat}, ${+long}), ll_to_earth(latitude, longitude)) AS distance
  , latitude 
  , longitude 
  , id
from Thing
```

Where `lat` and `long` are passed in via the request. With debug logging you'll get something like this:

```
  prisma:client prisma.queryRaw(
  prisma:client     select
  prisma:client     earth_distance(ll_to_earth($1, $2), ll_to_earth(latitude, longitude)) AS distance
  prisma:client     , latitude 
  prisma:client     , longitude 
  prisma:client     , id
  prisma:client     from Thing
  prisma:client     , [1,1])
```

Yep, that looks ok, but it will throw the following error:

```
db error: ERROR: value out of range: underflow
```

However, if the `lat` or a `long` is not a whole number, for example, `1.000001`, then the query works fine.

It feels like something is sending `1` as the wrong datatype??

I've tried the same scenario just using the Node Postgres driver.

```javascript
require('dotenv').config();
const { Pool, Client } = require('pg');

(async () => {
  //
  const client = new Client();

  await client.connect();

  const res = await client.query(
    `
select
  earth_distance(ll_to_earth(${+lat}, ${+long}), ll_to_earth(latitude, longitude)) AS distance
  , latitude 
  , longitude 
  , id
from thing
  `,
    [1, 1]
  );

  console.log('res', JSON.stringify(res.rows, null, 2));

  await client.end();
})().catch(console.error);
```

This works fine. No error.

I've also tried things like `${+lat}::float` but this does not help. 

An ugly hack is using `${+lat * 0.9999999999}`, which works fine if you pass `1`, but the error will return if you pass `1.0000000001`.

Any thoughts on what could be going wrong and how to work around it?

**Comment by aeddie-zapidhire** [comment]
Related to https://github.com/prisma/prisma-engines/issues/160 ??

**Comment by janpio** [comment]
Possible. Please report a bug instead of a discussion and provide all the information it asks for, then we can reproduce and take a look into fixing this.

**Comment by aeddie-zapidhire** [comment, marked as answer]
Issue -> https://github.com/prisma/prisma/issues/16611

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
